src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -266,6 +272,8 @@ public async Task ProcessRequestsAsync<TContext>(IHttpApplication<TContext> appl
                         stream.Abort(connectionError);
                     }
 
+                    await _streamsCompleted.Task;
```

Comment:
Timeout/cancellationToken?

Reply:
Is the abortive shutdown path via Abort(...) not sufficient?

Reply:
That's a good start, but right now you have no guarantee that the Task here will ever complete.

Reply:
I don't think we need to have a timeout or a CancellationToken here. What we have now is in line with the behavior for HTTP/1. If a connection's ProcessRequestsAsync method doesn't complete before the shutdown timeout CancellationToken fires, then CloseAllConnectionsAsync will complete and AbortAllConnectionsAsync will be called.

It's think it's a good idea to complete this TCS in Http2Connection.Abort() though.

Reply:
Hmm looks like the only impact if the task doesn't complete is if there are outstanding streams. After the graceful timeout, the abortive path takes over and still completes/aborts the _frameWriter. The only difference is that Input.Complete() is not called. Is that the concern?

Reply:
Now that I think about it more, I think it's a bad idea to complete this TCS in Abort. The socket get's closed in abort (if it isn't already), and that itself might unblock the application. AbortAllConnectionsAsync will timeout after a second if that's not the case anyway.